shared-bindings/time/__init__.c

```diff
@@ -52,7 +52,9 @@
 //|   :rtype: float
 //|
 STATIC mp_obj_t time_monotonic(void) {
-    return mp_obj_new_float(common_hal_time_monotonic() / 1000.0);
+    uint64_t time64 = common_hal_time_monotonic();
+    // 4294967296 = 2^32
+    return mp_obj_new_float(((uint32_t) (time64 >> 32) * 4294967296.0f + (uint32_t) (time64 & 0xffffffff)) / 1000.0f);
```

Comment:
How does precision change with this? Would we get more precision by dividing by 1000.0f earlier?

Reply:
I could distribute the division among the two halves, but the precision is still lost due to single-precision floating point. So I don't believe it will make a difference:
```
>>> (2.0**22+1)/1000 - (2.0**22+0)/1000 
0.0
>>> (2.0**22+2)/1000 - (2.0**22+1)/1000 
0.00195313
>>> (2.0**22+3)/1000 - (2.0**22+2)/1000 
0.0
>>> (2.0**22+4)/1000 - (2.0**22+3)/1000 
0.00195313
>>> 
```